python/sglang/srt/models/deepseek_v2.py

```diff
@@ -316,7 +333,14 @@ def forward_normal(self, hidden_states: torch.Tensor) -> torch.Tensor:
         if shared_output is not None:
             final_hidden_states = final_hidden_states + shared_output
         if self.tp_size > 1:
-            final_hidden_states = tensor_model_parallel_all_reduce(final_hidden_states)
+            if self.tp_size == self.dp_size:
+                tensor_list = list(final_hidden_states.tensor_split(split_indices_cpu))
```

Comment:
How about using `Tensor.split()` rather than `Tensor.tensor_split()`? This can reuse the info in `forward_batch.global_num_tokens_gpu` to make the code concise.